utilix/processing_requests.py

```diff
@@ -0,0 +1,152 @@
+import os
+from typing import Literal
+import rframe
+import datetime
+import getpass
+import pydantic
+import requests
+from warnings import warn
+
+CACHE = {}
+DEFAULT_ENV = '2022.03.5'
+ENV_TAGS_URL = 'https://api.github.com/repos/xenonnt/base_environment/git/matching-refs/tags/'
+
+API_URL = 'http://api.cmt.yossisprojects.com'
+
+
+def xeauth_user():
+    return os.environ.get('XEAUTH_USER', 'UNKNOWN')
+
+def get_envs():
+    r = requests.get(ENV_TAGS_URL)
+    if not r.ok:
+        return []
+    tags = r.json()
+    tagnames = [tag['ref'][10:] for tag in tags if tag['ref'] ]
+    return tagnames
+
+def default_env():
+    cond_env = os.environ.get('CONDA_DEFAULT_ENV', '')
+    if cond_env.startswith('XENONnT_'):
+        return cond_env.replace('XENONnT_', '')
+    envs = get_envs()
+    if envs:
+        return envs[-1]
+    return DEFAULT_ENV
+
+class ProcessingRequest(rframe.BaseSchema):
+    '''Schema definition for a processing request
+    '''
+    _NAME = 'processing_requests'
+
+    data_type: str = rframe.Index()
+    lineage_hash: str = rframe.Index()
+    run_id: str = rframe.Index()
+    destination:  Literal['OSG','DALI'] = rframe.Index(default='DALI')
+    user: str = pydantic.Field(default_factory=xeauth_user)
+    request_date: datetime.datetime = pydantic.Field(default_factory=datetime.datetime.utcnow)
+    
+    priority: int = -1
+
+    comments: str = ''
+    
+    
+    def pre_update(self, datasource, new):
+        if new.user != self.user:
+            raise ValueError(new.user)
+        if new.run_id != self.run_id:
+            raise ValueError(new.run)
+
+
+class ProcessingJob(rframe.BaseSchema):
+    _NAME = 'processing_jobs'
+
+    job_id: str = rframe.Index()
+    location: Literal['OSG','DALI','OTHER'] = rframe.Index()
```

Comment:
I think we should make these the actual RSE variables.

Reply:
good point

Reply:
done.